src/bundle.ts

```diff
@@ -32,14 +36,22 @@ export class BuildBundler extends Transform {
   bundler: Bundler;
   sharedFile: File;
 
+  files = new Map<string, File>();
+
   constructor(config: ProjectConfig, analyzer: BuildAnalyzer) {
     super({objectMode: true});
 
     this.config = config;
+
     this.analyzer = analyzer;
     this.sharedBundleUrl = 'shared-bundle.html';
     this.bundler = new Bundler({
-      analyzer: analyzer.analyzer,
+
+      // TODO(usergenic): Creating a new Analyzer with a blank cache is going
+      // to mean, at least, a doubling of analysis efforts for bundling phase.
+      // Ideally we would fork existing analyzer and replace its urlLoader if
+      // there were an affordance to do so.
+      analyzer: new Analyzer({urlLoader: new StreamLoader(this)}),
```

Comment:
I like the idea of using `_fork()` except that its still to early to rely on in this PR.

 - I don't have a good solution for `// do something to modify the url loader` yet without a public API change to `_fork()` or deep reaching into Analyzer's privates.
 - There needs to be some code to identify the document paths to invalidate based on which files have changed since original analysis or the cache will preserve the state of the original scanned scripts/stylesheets instead of getting the downstream (minified etc) versions during inlining.